lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/TraversalParams.java

```diff
@@ -0,0 +1,95 @@
+package com.kmwllc.lucille.connector.storageclient;
+
+import static com.kmwllc.lucille.connector.FileConnector.GET_FILE_CONTENT;
+import static com.kmwllc.lucille.connector.FileConnector.HANDLE_ARCHIVED_FILES;
+import static com.kmwllc.lucille.connector.FileConnector.HANDLE_COMPRESSED_FILES;
+import static com.kmwllc.lucille.connector.FileConnector.MOVE_TO_AFTER_PROCESSING;
+import static com.kmwllc.lucille.connector.FileConnector.MOVE_TO_ERROR_FOLDER;
+
+import com.kmwllc.lucille.core.fileHandler.FileHandler;
+import com.typesafe.config.Config;
+import java.net.URI;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+/**
+ * The arguments / parameters associated with a traversal on a StorageClient.
+ */
+public class TraversalParams {
+
+  // provided arguments
+  public final URI pathToStorageURI;
+  public final String docIdPrefix;
+  private final List<Pattern> excludes;
+  private final List<Pattern> includes;
+  public final Config fileOptions;
+
+  // derived arguments
+  public final boolean getFileContent;
+  public final String bucketOrContainerName;
+  public final boolean handleArchivedFiles;
+  public final boolean handleCompressedFiles;
+  public final String moveToAfterProcessing;
+  public final String moveToErrorFolder;
+  public final String startingDirectory;
+
+  public TraversalParams(URI pathToStorageURI, String docIdPrefix, List<Pattern> includes, List<Pattern> excludes, Config fileOptions, boolean forAzure) {
+    this.pathToStorageURI = pathToStorageURI;
+    this.docIdPrefix = docIdPrefix;
+    this.includes = includes;
+    this.excludes = excludes;
+    this.fileOptions = fileOptions;
+    this.getFileContent = !fileOptions.hasPath(GET_FILE_CONTENT) || fileOptions.getBoolean(GET_FILE_CONTENT);
+    this.handleArchivedFiles = fileOptions.hasPath(HANDLE_ARCHIVED_FILES) && fileOptions.getBoolean(HANDLE_ARCHIVED_FILES);
+    this.handleCompressedFiles = fileOptions.hasPath(HANDLE_COMPRESSED_FILES) && fileOptions.getBoolean(HANDLE_COMPRESSED_FILES);
+    this.moveToAfterProcessing = fileOptions.hasPath(MOVE_TO_AFTER_PROCESSING) ? fileOptions.getString(MOVE_TO_AFTER_PROCESSING) : null;
+    this.moveToErrorFolder = fileOptions.hasPath(MOVE_TO_ERROR_FOLDER) ? fileOptions.getString(MOVE_TO_ERROR_FOLDER) : null;
+
+    if (forAzure) {
```

Comment:
is there any way to move the special purpose Azure logic out of TraversalParams and into AzureStorageClient itself? In other words, could AzureStorageClient take the same bucketOrContainerName and startingDirectory as other clients, and do the necessary transformation of those values after accessing them from TraversalParams?